modules/apps/bookmarks/bookmarks-web/src/main/java/com/liferay/bookmarks/web/internal/portlet/action/ViewMVCRenderCommand.java

```diff
@@ -74,8 +74,9 @@ public String render(
 		return "/bookmarks/view.jsp";
 	}
 
-	@Reference
-	private BookmarksPortletToolbarContributor
-		_bookmarksPortletToolbarContributor;
+	@Reference(
+		target = "(component.name=com.liferay.bookmarks.web.internal.portlet.toolbar.contributor.BookmarksPortletToolbarContributor)"
+	)
+	private PortletToolbarContributor _bookmarksPortletToolbarContributor;
```

Comment:
Hey @MrJohn1911, We can remove the cast from [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/bookmarks/bookmarks-web/src/main/java/com/liferay/bookmarks/web/internal/display/context/BookmarksManagementToolbarDisplayContext.java#L112) safely. Please do that. Thx